go/connector/run.go

```diff
@@ -167,29 +204,71 @@ func runCommand(
 	var cmd = exec.Command(args[0], args[1:]...)
 	var fe = new(firstError)
 
-	// Copy |writeLoop| into connector stdin.
-	wc, err := cmd.StdinPipe()
-	if err != nil {
-		return fmt.Errorf("StdinPipe: %w", err)
-	}
-	go func() {
-		defer wc.Close()
-		fe.onError(writeLoop(wc))
-	}()
-
-	var stderrForwarder = ops.NewLogForwardWriter("connector stderr", logrus.InfoLevel, logger)
-
-	// Decode and forward connector stdout to |output|, but intercept a
+	// Decode and forward connector output from socket to |output|, but intercept a
 	// returned error to cancel our context and report through |fe|.
 	// If we didn't cancel, then the connector would run indefinitely.
-	cmd.Stdout = &writeErrInterceptor{
+	var outputInterceptor = &writeErrInterceptor{
 		delegate: output,
 		onError: func(err error) error {
 			fe.onError(err)
 			cancel() // Signal to exit.
 			return err
 		},
 	}
+
+	var conn net.Conn
+	// If port is specified, use TCP socket
+	if port != "" {
+		// Routine dialing a connection to connector
+		go func() {
+			// Try to connect to the connector with a retry mechanism
+			var connectionCounter = 0
+			var err error
+			for {
+				conn, err = net.Dial("tcp", fmt.Sprintf("localhost:%s", port))
+				if err != nil {
```

Comment:
This retry loop might be a bit clearer to read if this were inverted to `if err == nil { break }`. This would allow the rest of the logic to be less nested and removes the need for an explicit `continue`.